src/storage/db/snapshot.rs

```diff
@@ -264,24 +293,73 @@ pub async fn upload_to_s3(
             .to_string();
         let key = format!("{}/{}", upload_dir, file_name);
 
-        info!(key, "Uploading chunk to s3");
-        let byte_stream = ByteStream::from_path(entry.path()).await?;
-        let upload_result = s3_client
-            .put_object()
-            .key(key.clone())
+        let mut file = tokio::fs::File::open(entry.path()).await?;
+        let mut buffer = Vec::new();
+        file.read_to_end(&mut buffer).await?;
```

Comment:
We shouldn't read everything into memory, we're chunking by 5mb anyway, we should read 5mb at a time into the buffer

Reply:
done